File: briar-headless/src/main/java/org/briarproject/bramble/identity/OutputAuthor.java
package org.briarproject.bramble.identity;

import org.briarproject.bramble.api.identity.Author;
import org.briarproject.bramble.api.nullsafety.NotNullByDefault;

import javax.annotation.concurrent.Immutable;

@Immutable
@NotNullByDefault
@SuppressWarnings("WeakerAccess")
public class OutputAuthor {

	public final byte[] id;
	public final String name;
	public final byte[] publicKey;

	public OutputAuthor(Author author) {
		this.id = author.getId().getBytes();
		this.name = author.getName();
		this.publicKey = author.getPublicKey();
	}

}

File: briar-headless/src/main/java/org/briarproject/bramble/identity/OutputAuthor.kt
package org.briarproject.bramble.identity

import org.briarproject.bramble.api.identity.Author
import org.briarproject.briar.api.identity.AuthorInfo
import org.briarproject.briar.headless.json.JsonDict

fun Author.output() = JsonDict(
    "formatVersion" to formatVersion,
    "id" to id.bytes,
    "name" to name,
    "publicKey" to publicKey.encoded
)

fun AuthorInfo.Status.output() = name.toLowerCase()
